### PR TITLE
Chore: Asset-balances improvement

### PR DESCRIPTION
**Pull request by syan095**
## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have written sufficient tests.
- [x] I have written and tested required migrations.
- [x] I have updated documentation where appropriate.

## Summary
Minor improvement to the Asset-balances pallet
Added some new unit test cases.

### CODE REVIEW

File: state-chain/pallets/cf-asset-balances/src/tests.rs [hunk 1]
```diff
@@ -180,16 +192,23 @@ pub fn refund_validators_polkadot() {
...
 			payed_gas(
 				ForeignChain::Ethereum,
 				100,
 				ForeignChainAddress::Eth(sp_core::H160([i as u8; 20])),
 			);
 		}
+		for i in 254u8..=255u8 {
+			payed_gas(
+				ForeignChain::Ethereum,
+				500,
```
Comment (dandanlen) [inline state-chain/pallets/cf-asset-balances/src/tests.rs:205]: I'm not sure what the benefit of this is? We don't test anything related to the amount.

File: state-chain/pallets/cf-asset-balances/src/tests.rs [hunk 2]
```diff
@@ -231,3 +250,24 @@ pub fn do_not_refund_if_amount_is_too_low() {
 		assert_egress(0, None);
 	});
 }
+
+#[test]
+pub fn refund_validators_solana() {
+	new_test_ext().execute_with(|| {
+		const REFUND_AMOUNT: u128 = 1_000u128;
```
Comment (dandanlen) [inline state-chain/pallets/cf-asset-balances/src/tests.rs:257]: This test is testing the same case as refund_validators_btc - maybe we can write a generic test that just takes a `ForeignChain` as the argument?

File: state-chain/pallets/cf-asset-balances/src/tests.rs (hunk 1 above)
Comment (syan095) [reply state-chain/pallets/cf-asset-balances/src/tests.rs:205]: Oh this is to test that the payouts are settled in reverse order (smallest to largest) so the 2 largest fees are left when we run out of available fund. I'll add a comment for this.

**Review by dandanlen — APPROVED** [review]
Thanks Roy

**Pull request merged**